http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -44,19 +53,106 @@
      */
     public static final SucceededFuture<?> SUCCEEDED_FUTURE = new SucceededFuture<>(null, null);
 
-    // TODO: add a link to the guide on how to diagnose this error here once it's available
-    public static final String CLOSED_CHANNEL_MESSAGE = "The channel was closed. This may have been done by the client (e.g. "
-                                                        + "because the request was aborted), " +
-                                                        "by the service (e.g. because there was a handshake error, the request "
-                                                        + "took too long, or the client tried to write on a read-only socket), " +
-                                                        "or by an intermediary party (e.g. because the channel was idle for too"
-                                                        + " long).";
-
     private static final Logger log = Logger.loggerFor(NettyUtils.class);
 
     private NettyUtils() {
     }
 
+    public static Throwable decorateException(Channel channel, Throwable originalCause) {
```

Comment:
Nit: We almost have enough exception-related logic in this class to consider a `NettyExceptionUtils` or similar.